File: Backend/src/main/java/com/jp/mainichishinpo/entity/Exam.java
package com.jp.mainichishinpo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "exam")
public class Exam {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Size(max = 50)
    private String exam_name;

    @ManyToMany(fetch = FetchType.LAZY, cascade = {CascadeType.DETACH, CascadeType.REFRESH})
    @JoinTable(	name = "exam_question",
            joinColumns = @JoinColumn(name = "exam_id"),
            inverseJoinColumns = @JoinColumn(name = "question_id"))
    private Set<Question> questions = new HashSet<>();

    private String note;

    private Boolean active;
    private String type;
    private Boolean hasAudio;
    private String url_audio;

    @OneToMany(mappedBy = "id", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Result> results;
}

File: frontend_v2/src/app/entity/Exam.ts
import {Question, QuestionFillWord} from "../entity/Question";
export class Exam{
  id:number = 0;
  exam_name: string | undefined;
  note: string |  undefined;
  active: boolean | undefined;
  listQuestionId: number[] | undefined;
  questions!: Question[] | undefined;
}

export class ExamFillWord{
  id:number = 0;
  exam_name: string | undefined;
  note: string |  undefined;
  active: boolean | undefined;
  listQuestionId: number[] | undefined;
  questions!: QuestionFillWord[] | undefined;
}

export class ExamDto{
  id:number = 0;
  exam_name: string | undefined;
  note: string |  undefined;
  active: boolean | undefined;
  questionTotal: number| undefined;
  type: string|undefined;
}
